lib/puppet/functions/conjur/secret.rb

```diff
@@ -13,7 +13,8 @@
   #   - account: Name of the Conjur account that contains this variable.
   #   - authn_login: The identity you are using to authenticate to the Conjur / DAP instance.
   #   - authn_api_key: The API key of the identity you are using to authenticate with.
-  #   - ssl_certificate: The _raw_ PEM-encoded x509 CA certificate chain for the DAP instance.
+  #   - cert_file: The absolute path to CA certificate chain for the DAP instance on the agent. This variable overrides `ssl_certificate`.
```

Comment:
`cert_file` is only implemented in the config options for Windows (regedit) and Linux (conjurrc). I think we need to add a little bit of code so that it's implemented for the `conjur::secret` method. It's pretty much duplicated functionality so that when `cert_file` is present we read the contents of `cert_file` and use those instead `ssl_certificate`.

Something like the code below, before [this](https://github.com/cyberark/conjur-puppet/blob/105-cert-file/lib/puppet/functions/conjur/secret.rb#L102-L106) should do.
```
if opts['cert_file']
  raise "Cert file '#{c['cert_file']}' cannot be found!" unless File.file?(opts['cert_file'])
  opts['ssl_certificate'] ||= File.read opts['cert_file']
end
```

Reply:
I force pushed these changes so no need to worry about these.